### ISSUE TITLE
Installation fails on nixos-unstable due to missing attribute ghc7103

### ISSUE THREAD

**Issue by dkellner** [eta-lang/eta-nix#2]
GHC 7.10.3 was dropped in `nixpkgs` about 2 months ago: https://github.com/NixOS/nixpkgs/commit/2f0de54ddbfdb03540128f8d2abcc47221d42b25

What is the best way to move forward? Should we provide another overlay for installing GHC 7.10.3?

**Comment by puffnfresh** [comment]
Oh that's a bugger!!

**Comment by puffnfresh** [comment]
An alternative would be to work on Eta to allow it to compile with GHC 8. Things like the Generic instances would have to be written because it uses that for communication. It's a chunk of work, but it would be good work.

**Comment by dkellner** [comment]
Of course, Eta supporting GHC 8 would be the best for various reasons :-).

Meanwhile we could at least update the installation docs to include this known limitation for `nixos-unstable`? I think 18.09 still includes GHC 7.10.3, so it will take until 19.03 until there's a problem using a stable release.

I'm not sure about the recommended remedy we should include in the docs... the Docker image is working, and using the stable Nix expressions from 18.09, too, but takes forever to compile:

```
NIXPKGS="https://github.com/NixOS/nixpkgs-channels/archive/nixos-18.09.tar.gz" \
NIX_PATH=nixpkgs=${NIXPKGS} \
nix-shell -p 'etaPackages.etaWithPackages (p: [ p.lens ])'
```

**Comment by apeyroux** [comment]
This problem still exists with the 19.03

**Comment by emberdart** [comment]
Yes this is now a problem in stable - so following the instructions no longer work at all.
Best to migrate to ghc 9 or at least 8.10 for now.
